rpcs3/Emu/Cell/Modules/cellSaveData.cpp

```diff
@@ -275,18 +271,23 @@ static error_code display_callback_result_error_message(ppu_thread& ppu, vm::ptr
 		msg = "Error - Save data cannot be found";
 		break;
 	case CELL_SAVEDATA_CBRESULT_ERR_INVALID:
-		if (result->invalidMsg)
+		if (result.invalidMsg)
 			use_invalid_message = true;
 		break;
 	default:
-		return CELL_SAVEDATA_ERROR_CBRESULT;
+		// ****** sysutil savedata parameter error : 22 ******
+		return {CELL_SAVEDATA_ERROR_PARAM, "22"};
 	}
 
+	// TODO: errDialog == CELL_SAVEDATA_ERRDIALOG_NOREPEAT
+	if (errDialog != CELL_SAVEDATA_ERRDIALOG_ALWAYS)
```

Comment:
was this change of order tested?

Reply:
What do you mean? I changed it so it always checks for totally invalid/inappropriate callback results to return error param instead of cbresult error.
it assumes that if we didn't check other values such as OK_LAST or OK_LAST_NOCONFIRM before it, they are treated as invalid values.

Afaik after all savedata callbacks in fw theres a switch case with 7 cases and 1 default path, 5 of the usual negative errors, 2 for OK_LAST + OK_LAST_NOCONFIRM, one for OK_NEXT and a default which always returns error param.